File: lua/problem_90.lua
local M = {}

function M.random_excluding(n, exclude)
    local excl = {}
    for _, v in ipairs(exclude) do excl[v] = true end
    local valid = {}
    for i = 0, n - 1 do
        if not excl[i] then valid[#valid+1] = i end
    end
    return valid[math.random(1, #valid)]
end

function M.problem90()
    print("Problem 90:")
    for _ = 1, 100 do
        assert(M.random_excluding(5, {0,1,2,4}) == 3)
    end
    local counts = {}
    for _ = 1, 1000 do
        local v = M.random_excluding(5, {2})
        counts[v] = (counts[v] or 0) + 1
    end
    for _, k in ipairs({0,1,3,4}) do
        assert((counts[k] or 0) > 100, "bad distribution")
    end
    print("All Tests Passed")
end

return M

File: typescript/problem_90.ts
export function randomExcluding(n: number, exclude: number[]): number {
    const excSet = new Set(exclude);
    const valid = Array.from({length:n},(_,i)=>i).filter(i=>!excSet.has(i));
    return valid[Math.floor(Math.random()*valid.length)];
}

export function problem90(): void {
    console.log("Problem 90:");
    for (let i=0;i<100;i++) if(randomExcluding(5,[0,1,2,4])!==3) throw new Error("assertion failed");
    const counts:Record<number,number>={};
    for(let i=0;i<1000;i++){const v=randomExcluding(5,[2]);counts[v]=(counts[v]??0)+1;}
    for(const k of [0,1,3,4]) if((counts[k]??0)<100) throw new Error("bad distribution");
    console.log("All Tests Passed");
}
